### PR TITLE
feat(platonic/regress): permutation p-values, pairwise MKNN/Wasserstein, UMAP

### PR DESCRIPTION
**Pull request by ksd3**
Adds the rest of Ashod's pipeline to \`experiments/platonic/regress/\`, plus a real significance test on the linear probes and UMAP coords ready for plotting.

## What's new

| Addition | Where |
|---|---|
| Permutation p-value on each linear probe (Phipson–Smyth corrected) | per-tuple, in \`pu_regress.py\` |
| Per-tuple cosine kNN graph (faiss, k=10 by default) | per-tuple, saved as \`done/<tag>/neighbours.parquet\` |
| Per-tuple UMAP-2D using the precomputed kNN graph | per-tuple, saved as \`done/<tag>/umap.parquet\` |
| Pairwise **MKNN overlap** between embeddings (intramodal + crossmodal) | new aggregator \`pu_regress_pairs.py\` |
| Pairwise **Wasserstein distance** per physics property | new aggregator \`pu_regress_pairs.py\` |

## Per-tuple output schema

\`done/<tag>/\` now contains three parquets uploaded as one \`upload_folder\` commit:

- \`probe.parquet\` — one row per property: \`r2_mean, r2_std, r2_pvalue, n, d, cv_folds, pca_k, n_perm\`
- \`neighbours.parquet\` — one row per galaxy: \`(k,)\` int32 indices in this embedding space
- \`umap.parquet\` — one row per galaxy: \`umap_x, umap_y\` (precomputed-kNN UMAP, cosine)

## Pairwise output schema

\`pu_regress_pairs.py\` pulls every neighbour matrix from HF and computes:

- **Intramodal:** adjacent sizes within a family on the same modality (e.g., \`(hsc, vit-base)\` vs \`(hsc, vit-large)\`).
- **Crossmodal:** \`(hsc, model_size)\` vs \`(jwst, model_size)\`.

Each pair gets one row with: \`pair_kind, modality_a/b, model_alias, size_a/b, n, k, mknn\` plus \`wass_<property>\` for each of the 6 physical labels.

## Mapping to Ashod's pipeline

Step-for-step coverage:

| Ashod | This PR |
|---|---|
| Catalog pass | \`pu_regress.collect_catalog\` |
| Embedding extraction per (model, size, modality) | \`pu_regress.extract_embeddings\` (cached) |
| Linear probe → R² | \`pu_regress._probe_with_pvalue\` (+ p-value, new) |
| kNN graph per embedding | \`pu_regress._build_knn\` (faiss/sklearn) |
| MKNN overlap intramodal + crossmodal | \`pu_regress_pairs.mknn_overlap\` |
| Wasserstein per property | \`pu_regress_pairs.wasserstein_per_property\` |
| Plots | (out of scope — UMAP coords saved so plotting is one short script) |

## Anonymisation / generality

Same pattern as the rest of \`experiments/platonic/\`. All knobs via env vars; no hard-coded HF tokens, namespaces, paths, or accounts. Scripts fail loudly if any required env is unset.

## Tested

End-to-end on a local RTX 5070 with N=300 / k=10 / B=20:

\`\`\`
[claim] hsc__dinov3_vits16 (facebook/dinov3-vits16-pretrain-lvd1689m, params_M=22)
catalog: 300 rows, 6 fields
extracted (300, 384), cached
building cosine kNN graph (k=10)
computing UMAP-2D (precomputed kNN)
  R²(redshift) = -1.71 ± 0.85  p=0.0476
  R²(mag_g)    = 0.39  ± 0.09  p=0.0476
  R²(mag_r)    = 0.44  ± 0.14  p=0.0476
  R²(mass)     = -1.74 ± 0.27  p=0.0476
  R²(sSFR)     = -1.39 ± 0.54  p=0.0476
  R²(g-r)      = 0.14  ± 0.08  p=0.0476
[done] hsc__dinov3_vits16 -> done/hsc__dinov3_vits16/{probe,neighbours,umap}.parquet
\`\`\`

UMAP parquet: \`(300, 3)\` with finite \`(umap_x, umap_y)\`.
Pairs aggregator on two tuples produces 1 intramodal pair: \`MKNN=0.36\`, all 6 Wasserstein distances populated.

### CODE REVIEW

**Comment by Shash12-ship** [comment]
I think if you increase the number of samples the p-values will be < 0.05.

**Review by Shash12-ship — APPROVED** [review]
